Spigot-API-Patches/0154-Add-Material-Tags.patch

```diff
@@ -613,6 +613,77 @@ index 0000000000000000000000000000000000000000..e72fc52240015051fe58b246fcf5c014
 +        .add(Material.LANTERN, Material.SOUL_LANTERN)
 +        .ensureSize("LANTERNS", 2);
 +
++    /**
++     * Covers the variants of rails.
++     */
++    public static final MaterialSetTag RAILS = new MaterialSetTag(keyFor("rails"))
++        .endsWith("RAIL")
++        .ensureSize("RAILS", 4);
++
++    /**
++     * Covers the variants of swords.
++     */
++    public static final MaterialSetTag SWORDS = new MaterialSetTag(keyFor("swords"))
++        .endsWith("_SWORD")
++        .ensureSize("SWORDS", 6);
++
++    /**
++     * Covers the variants of shovels.
++     */
++    public static final MaterialSetTag SHOVELS = new MaterialSetTag(keyFor("shovels"))
++        .endsWith("_SHOVEL")
++        .ensureSize("SHOVELS", 6);
++
++    /**
++     * Covers the variants of pickaxes.
++     */
++    public static final MaterialSetTag PICKAXES = new MaterialSetTag(keyFor("pickaxes"))
++        .endsWith("_PICKAXE")
++        .ensureSize("PICKAXES", 6);
++
++    /**
++     * Covers the variants of axes.
++     */
++    public static final MaterialSetTag AXES = new MaterialSetTag(keyFor("axes"))
++        .endsWith("_AXE")
++        .ensureSize("AXES", 6);
++
++    /**
++     * Covers the variants of hoes.
++     */
++    public static final MaterialSetTag HOES = new MaterialSetTag(keyFor("hoes"))
++        .endsWith("_HOE")
++        .ensureSize("HOES", 6);
++
++    /**
++     * Covers the variants of helmets.
++     */
++    public static final MaterialSetTag HELMETS = new MaterialSetTag(keyFor("helmets"))
++        .endsWith("_HELMET")
++        .ensureSize("HELMETS", 7);
++
++    /**
++     * Covers the variants of items that can be equipped in the chestplate slot.
++     */
++    public static final MaterialSetTag CHESTPLATES = new MaterialSetTag(keyFor("chestplates"))
++        .endsWith("_CHESTPLATE")
++        .add(Material.ELYTRA)
```

Comment:
Didn't know an elytra is a chestplate 👀

Reply:
https://minecraft.gamepedia.com/Chestplate
They aren't :p

Reply:
After some thinking, it's probably best to have two tags:
CHEST_SLOT, and CHESTPLATE (or an alternative name for the former).
It isn't instantly obvious that an elytra would be a *chestplate* but it would be helpful to know which materials can be used in the chest *slot*.

Reply:
Same should go with turtle shell in helm slot.

https://minecraft.gamepedia.com/Turtle_Shell

Reply:
The Turtle Shell is already included, its material is TURTLE_HELMET and so it automatically gets added to helmets with the rest of them.